app/models/manageiq/providers/kubevirt/infra_manager/provision/cloning.rb

```diff
@@ -54,4 +78,60 @@ def user_options(options)
     merged_options[:cloud_user_password] = ManageIQ::Password.try_decrypt(get_option(:root_password))
     merged_options.compact
   end
+
+  def values(template, options)
+    default_params = {}
+    template.parameters.each do |param|
+      name = param[:name].downcase
+      value = options[name.to_sym]
+      if value && name == "memory"
+        value = "#{value}Mi"
+      end
+
+      default_params[name] = value || param[:value]
+    end
+    default_params
+  end
+
+  def param_substitution!(object, params)
+    result = object
+    case result
+    when Hash
+      result.each do |k, v|
+        result[k] = param_substitution!(v, params)
+      end
+    when Array
+      result.map { |v| param_substitution!(v, params) }
+    when String
+      result = sub_specific_object(params, object)
+    end
+    result
+  end
+
+  #
+  # Performs substitution on specific object.
+  #
+  # @params params [Hash] Containing parameter names and values used for substitution.
+  # @params object [String] Object on which substitution takes place.
+  # @returns [String] The outcome of substitution.
+  #
+  def sub_specific_object(params, object)
```

Comment:
Method name is weird to me - thinking `substitute_string` or `substitute_literal`.  Not important for merge.

Reply:
This came from fog-kubevirt https://github.com/fog/fog-kubevirt/blob/master/lib/fog/kubevirt/compute/models/template.rb#L189-L206